app/controllers/shelter_reviews_controller.rb

```diff
@@ -1,3 +1,37 @@
 class ShelterReviewsController < ApplicationController
-  
+
+  def new
+    @shelter = Shelter.find(params[:id])
+  end
+
+  def create
+    shelter_id = params[:id]
+    if params[:title].empty? || params[:rating].empty? || params[:content].empty?
+      messages = []
+      messages << "Please fill out the following fields:"
+      messages << "Title" if params[:title].empty?
+      messages << "Rating" if params[:rating].empty?
+      messages << "Content" if params[:content].empty?
+      flash[:error] = messages
+      redirect_to "/shelter_reviews/#{shelter_id}/new"
```

Comment:
I like how dynamic this is. I think we should pull out some of this logic into a private helper method.